setup.py

```diff
@@ -83,19 +83,43 @@ class BuildUI(Command):
         ("clean", "c", "clean directory"),
     ]
 
+    @staticmethod
+    def _node_versions():
+        import requests
+
+        r = requests.get("https://nodejs.org/download/release/index.json")
+        return r.json()
+
+    @staticmethod
+    def _latest_lts(node_versions):
+        for version in node_versions:
+            if version["lts"]:
+                return version["version"]
+
+    @staticmethod
+    def _node_lts(node_versions):
+        import subprocess
+
+        proc = subprocess.Popen(["node", "-v"], stdout=subprocess.PIPE)
+        installed_version = proc.stdout.read().decode("utf-8")[:-1]
+        print(f"The installed Node.js version is {installed_version}.")
+        for version in node_versions:
+            if version["version"] == installed_version:
+                return version["lts"]
+
     def initialize_options(self):
         self.clean = False
 
     def finalize_options(self):
         pass
 
     def run(self):
+        node_versions = self._node_versions()
         if self.clean:
             import shutil
 
             shutil.rmtree("covalent_ui/webapp/build", ignore_errors=True)
-
-        else:
+        elif self._node_lts(node_versions):
             import subprocess
 
             subprocess.run(
```

Comment:
for each of these subprocess.run commands, we should add this:
```
if proc.returncode != 0:
    raise Exception(proc.stderr.decode("utf-8").strip())
```
where `proc` is returned from `subprocess.run`.  Otherwise we might miss errors in the subprocess.